components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -169,10 +170,31 @@ func TheiaReadyHandler(r *mux.Router, config *RouteHandlerConfig) {
 
 // TheiaSupervisorReadyHandler handles /supervisor/ready
 func TheiaSupervisorReadyHandler(r *mux.Router, config *RouteHandlerConfig) {
+	// We MUST NOT proxy-pass to the workspace-internal supervisor endpoint.
+	// There's a lot going on there that's not supposed to be available from outside.
 	r.Use(config.CorsHandler)
-	r.NewRoute().
-		HandlerFunc(proxyPass(config, workspacePodSupervisorResolver,
-			withOnProxyErrorRedirectToWorkspaceStartHandler(config.Config)))
+	r.NewRoute().HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		url, err := workspacePodSupervisorResolver(config.Config, req)
+		if err != nil {
+			log.WithError(err).Error("cannot answer supervisor/ready call")
+			http.Error(resp, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+
+		timeout := 2 * time.Second
```

Comment:
Does it mean that we loose at most 2 secs on the startup time here?

Reply:
If everything goes according to plan (i.e. supervisor signals ready at some point) we shouldn't loose any time here. In case there's a network fluke we'd loose the 2 seconds worst case, indeed.

That said, I reckon the two seconds are way too aggressive because the ready request made here is meant to be long running until the content is available. We' might want to up this value down the road to something like 30 sec.